src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -788,6 +791,27 @@ static term gpiodriver_set_int(Context *ctx, int32_t target_pid, term cmd)
         return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_trigger_atom));
     }
 
+    if (term_get_tuple_arity(cmd) == 4) {
+        term pid = term_get_tuple_element(cmd, 3);
+        if (UNLIKELY(!term_is_pid(pid) && !term_is_atom(pid))) {
+            AVM_LOGE(TAG, "Invalid listener parameter, must be a pid() or registered process!");
+            return create_pair(ctx, ERROR_ATOM, invalid_listener_atom);
```

Comment:
I don't see any memory checks (memory_ensure_free*) for the tuple here (or in the above calls to create_pair).  We are probably getting lucky because by default we allocate a little bit of headroom in the heap, but this could still result in a crash in some cases.